### PR TITLE
feat: add conversion for fixed ABI types

### PR DESCRIPTION
**Pull request by BowTiedDevil**
### What I did

Ape has no converter to handle Vyper's `decimal` type, which is translated to the `fixed168x10` ABI type. Testing a contract function with a `decimal` type argument will fail with a `ConversionError`.

#### `ape_ethereum` plugin:
- Add `Decimal` as the native Python type for ABI types with "fixed" in the name.

#### `ape` base:
- Add a converter to support using string-formatted float values instead of requiring a strictly `Decimal` value. This converter will translate a "1.0" string in-place. Did not add a `float` converter, because floating-point accuracy will cause subtle errors. 

#### Tests:
- Added simple string -> Decimal converter test.

### Checklist:
- [ ] All changes are completed
- [x] New test cases have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: src/ape/managers/converters.py [hunk 1]
```diff
@@ -185,6 +186,20 @@ def convert(self, value: Union[str, datetime, timedelta]) -> int:
...
 
+class StringDecimalConverter(ConverterAPI):
+    """
+    Convert string-formatted floating point values to `Decimal` type.
+    """
+
+    def is_convertible(self, value: Any) -> bool:
+        # Matches only string-formatted floats with an optional sign character (+/-),
+        # leading and trailing zeros are optional.
+        return isinstance(value, str) and re.fullmatch("[+-]?[0-9]*[.][0-9]*", value)
```
Comment (fubuloubu) [inline src/ape/managers/converters.py:197]: Only nit is if theres a simpler regex to run here that picks up decimal values

Or even just trying to cast to decimal and catching exceptions

**Review by fubuloubu — APPROVED** [review]

File: src/ape/managers/converters.py (hunk 1 above)
Comment (BowTiedDevil) [reply src/ape/managers/converters.py]: I've reworked the regex to require digits before and after the decimal, because a "." input was slipping through.

Comment (fubuloubu) [reply src/ape/managers/converters.py]: using `\d` might even be clearer (lots of alpha [here](https://docs.python.org/3/howto/regex.html))

Comment (fubuloubu) [reply src/ape/managers/converters.py]: also make sure to use `\.` and not just `.` (which is "any character")

**Comment by fubuloubu** [comment]
Also, general testing advice, using a regex is usually great paired with fuzzing to find corner cases for your expression (given positive and negative examples of what you want it to handle)

File: src/ape/managers/converters.py (hunk 1 above)
Comment (BowTiedDevil) [reply src/ape/managers/converters.py]: That's interesting, I used \d and got a depreciation warning from the re module. I didn't look further into it, but will take another look now.

Comment (BowTiedDevil) [reply src/ape/managers/converters.py]: Resolved with ba341bb

**Review by fubuloubu — DISMISSED** [review]

File: tests/functional/conversion/test_decimal.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+from decimal import Decimal
+
+import pytest
+
+
+def test_converting_formatted_float_strings_to_decimal(convert):
```
Comment (antazoey) [inline tests/functional/conversion/test_decimal.py:6]: nit: I tend to use `-k` to run multiple tests and it works great if following this pattern:

```
test_<method-name>_<rest>`
```

```suggestion
def test_convert_formatted_float_strings_to_decimal(convert):
```

**Review by antazoey — APPROVED** [review]

File: src/ape/managers/converters.py [hunk 3]
```diff
@@ -185,6 +186,20 @@ def convert(self, value: Union[str, datetime, timedelta]) -> int:
...
 
+class StringDecimalConverter(ConverterAPI):
+    """
+    Convert string-formatted floating point values to `Decimal` type.
+    """
+
+    def is_convertible(self, value: Any) -> bool:
+        # Matches only string-formatted floats with an optional sign character (+/-).
+        # Leading and trailing zeros are required.
+        return isinstance(value, str) and re.fullmatch(r"[+-]?\d+\.\d+", value) is not None
```
[the remark this answers is not in the document]
Comment (BowTiedDevil) [reply src/ape/managers/converters.py:199]: Confirmed, `re.fullmatch` will check the full string - it _should_ be equivalent to sandwiching the above expression with `^...$` but I'm not experienced enough with regex to know for sure if that expression has edge cases.

Added a note to the comment with b31222a

File: tests/functional/conversion/test_decimal.py (hunk 2 above)
Comment (BowTiedDevil) [reply tests/functional/conversion/test_decimal.py]: I reformatted the test names per your suggestion, and added some more to check other bad strings per @fubuloubu's comment above.

File: tests/functional/conversion/test_decimal.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+            convert(test_string, Decimal)
+
+
+def test_convert_strings_with_token_names(convert):
+    test_strings = [
+        "0.999 DAI",
+        "10.0 USDC",
+    ]
+    for test_string in test_strings:
+        with pytest.raises(Exception):
```
Comment (antazoey) [inline tests/functional/conversion/test_decimal.py:44]: The actual exception should go in these spots... ConversionError?

Comment (BowTiedDevil) [reply tests/functional/conversion/test_decimal.py]: Good call, I've swapped `Exception` to `ConversionError` and included a pattern match in 99f61b8

**Review by antazoey — APPROVED** [review]

**Review by fubuloubu — APPROVED** [review]
